VertexAISnippets/VertexAISnippets/VertexAISnippets.swift

```diff
@@ -0,0 +1,329 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(UIKit)
+import UIKit
+#else
+import AppKit
+#endif
+
+// [START import_vertexai]
+import FirebaseVertexAI
+// [END import_vertexai]
+
+class Snippets {
+
+  var model: GenerativeModel!
+
+  func initializeModel() {
+    // [START initialize_model]
+    // Initialize the Vertex AI service
+    let vertex = VertexAI.vertexAI()
+
+    // Initialize the generative model with a model that supports your use case
+    // Gemini 1.5 Pro is versatile and can accept both text-only or multimodal prompt inputs
+    let model = vertex.generativeModel(modelName: "gemini-1.5-pro-preview-0409")
+    // [END initialize_model]
+
+    self.model = model
+  }
+
+  func callGemini() async throws {
+    // [START call_gemini]
+    // Provide a prompt that contains text
+    let prompt = "Write a story about a magic backpack."
+
+    // To generate text output, call generateContent with the text input
+    let response = try await model.generateContent(prompt)
+    if let text = response.text {
+      print(text)
+    }
+    // [END call_gemini]
+  }
+
+  func callGeminiStreaming() async throws {
+    // [START call_gemini_streaming]
+    // Provide a prompt that contains text
+    let prompt = "Write a story about a magic backpack."
+
+    // To stream generated text output, call generateContentStream with the text input
+    let contentStream = model.generateContentStream(prompt)
+    for try await chunk in contentStream {
+      if let text = chunk.text {
+        print(text)
+      }
+    }
+    // [END call_gemini_streaming]
+  }
+
+  func sendTextOnlyPromptStreaming() async throws {
+    // [START text_only_prompt_streaming]
+    // Provide a prompt that contains text
+    let prompt = "Write a story about a magic backpack."
+
+    // To stream generated text output, call generateContentStream with the text input
+    let contentStream = model.generateContentStream(prompt)
+    for try await chunk in contentStream {
+      if let text = chunk.text {
+        print(text)
+      }
+    }
+    // [END text_only_prompt_streaming]
+  }
+
+  // Note: This is the same as the call gemini prompt, but may change in the future.
```

Comment:
I think this is the one that we want to keep and remove the "call gemini" one above.

Reply:
Ack